x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/aggregate/HasPercentilesConfig.java

```diff
@@ -0,0 +1,189 @@
+            }
+
+            // if method is null, the method parameter is not checked
+            if (methodParameter != null && Expressions.isNull(methodParameter) == false) {
+                resolution = isFoldable(methodParameter, sourceText, methodParameterOrdinal);
+                if (resolution.unresolved()) {
+                    return resolution;
+                }
+
+                MethodParameterResolver resolver = METHOD_TO_RESOLVER.get(percentilesMethod);
+                if (resolver == null) {
+                    // so in the future if a new method is added, at least the users will be able to use it with 
+                    // the default parameters, but won't be able to configure it until the resolver is added
+                    return new Expression.TypeResolution(format(null,
+                        "the [{}] method can only be used with the default method parameters, please omit the {} argument of [{}]",
+                        methodParameterOrdinal.name().toLowerCase(Locale.ROOT), sourceText, methodName));
+                }
+                resolution = resolver.resolve(methodParameter, sourceText, methodParameterOrdinal);
+                return resolution;
+            }
+        }
+
+        return Expression.TypeResolution.TYPE_RESOLVED;
+    }
+
+    private static PercentilesConfig asPercentilesConfig(Expression method, Expression methodParameter) {
+        if (method == null) {
+            // sadly we had to set the default here, the PercentilesConfig does not provide a default
+            return new PercentilesConfig.TDigest();
+        }
+        String methodName = foldOptionalNullable(method, DataTypes.KEYWORD);
+        PercentilesMethod percentilesMethod = null;
+        for (PercentilesMethod m : PercentilesMethod.values()) {
+            if (m.getParseField().getPreferredName().equals(methodName)) {
+                percentilesMethod = m;
+                break;
+            }
+        }
+        if (percentilesMethod == null) {
+            throw new IllegalStateException("Not handled PercentilesMethod [" + methodName + "], type resolution needs fix");
+        }
+        switch (percentilesMethod) {
+            case TDIGEST:
+                Double compression = foldOptionalNullable(methodParameter, DataTypes.DOUBLE);
+                return compression == null ? new PercentilesConfig.TDigest() : new PercentilesConfig.TDigest(compression);
+            case HDR:
+                Integer numOfDigits = foldOptionalNullable(methodParameter, DataTypes.INTEGER);
+                return numOfDigits == null ? new PercentilesConfig.Hdr() : new PercentilesConfig.Hdr(numOfDigits);
+            default:
+                throw new IllegalStateException("Not handled PercentilesMethod [" + percentilesMethod + "], type resolution needs fix");
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private static <T> T foldOptionalNullable(Expression e, DataType dataType) {
```

Comment:
The name is misleading since nullable means something else. rename to `foldNullSafe`